### PR TITLE
Added libraries for validate

### PR DESCRIPTION
**Pull request by chitrak7**

### CODE REVIEW

**Comment by chitrak7** [comment]
Added libraries check-ppr and check-api-annotations to stage2 packages.

File: src/GHC.hs [hunk 1]
```diff
@@ -199,7 +201,7 @@ stage1Packages = do
           ++ [ win32    | win     ]
 
 stage2Packages :: Action [Package]
-stage2Packages = return [haddock]
+stage2Packages = return [haddock, checkPpr, checkApiAnnotations]
```
Comment (snowleopard) [inline src/GHC.hs]: We don't want to build these two packages by default, so they shouldn't be added here.

Instead, `need` then in the `validate` rule.

**Comment by snowleopard** [comment]
Thanks @chitrak7! See my comment above. Adding these two utilities into the default build will slow down Hadrian unnecessarily. We only want to build them in the `validate` rule, before running the testsuite.

File: src/GHC.hs (hunk 1 above)
Comment (chitrak7) [reply src/GHC.hs]: I think that intead, I can export packages as an [package] and make it as a need for  validate rule. It should do this for us.

Comment (snowleopard) [reply src/GHC.hs]: @chitrak7 Are you saying you'd like to do something like this?
```haskell
-- In GHC.hs
testsuitePackages :: [Package]
testsuitePackages = [checkPpr, checkApiAnnotations]

-- Rules/Test.hs
...
needPackages testsuitePackages
...
```
I think I like this. You'll need to define a helper function `needPackages` for that.

Comment (chitrak7) [reply src/GHC.hs]: Yes, exactly what i wish to do. Will implement this after recent pulls are merged

Comment (snowleopard) [reply src/GHC.hs]: Great, thank you!

**Comment by snowleopard** [comment]
@chitrak7 Can you rebase this PR and implement the above suggestion? I'd like to merge this so we can reduce the number of `validate` failures.

**Comment by chitrak7** [comment]
@snowleopard Can I do this task on 1st as I have my end semesters going on now.

**Comment by snowleopard** [comment]
@chitrak7 Sure, no problem!

This is a small fix so if we get blocked on this, we'll just complete this PR ourselves and merge it.

**Pull request closed without merging**